met-web/src/components/engagement/view/PreviewBanner.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext } from 'react';
+import { ActionContext } from './ActionContext';
+import { Box, Button, Grid, Skeleton, Typography } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { EngagementStatus } from 'constants/engagementStatus';
+import { ConditionalComponent } from 'components/common';
+import { useAppSelector } from 'hooks';
+
+export const PreviewBanner = () => {
+    const navigate = useNavigate();
+    const { engagementLoading, savedEngagement, publishEngagement } = useContext(ActionContext);
+    const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+
+    const engagementId = savedEngagement.id || '';
+    if (engagementLoading) {
+        return <Skeleton variant="rectangular" width="100%" height="10em" />;
+    }
+    const isDraft = savedEngagement.status_id === EngagementStatus.Draft;
+
+    const validate = () => {
+        const errors = {
+            status: savedEngagement.status_id !== EngagementStatus.Draft,
+        };
+
+        return Object.values(errors).some((isError: unknown) => isError);
+    };
+
+    const handlePublishEngagement = async () => {
+        const hasErrors = validate();
+
+        if (!hasErrors) {
+            const engagement = await publishEngagement({
+                ...savedEngagement,
+                status_id: EngagementStatus.Published,
+                published_date: new Date().toUTCString(),
+                surveys: [],
+            });
+
+            navigate(`/`);
+
+            return engagement;
+        }
+    };
+
+    return (
+        <ConditionalComponent condition={isDraft && isLoggedIn}>
+            <Box
+                sx={{
+                    backgroundColor: 'secondary.light',
+                }}
+            >
+                <Grid
+                    container
+                    direction="column"
```

Comment:
This is obviously looking at the snapshot already working. I just have a feeling here that you might not need the direction='column', row perhaps would be better. The grid props like xs={12} will work only for a row direction since it is a 12 column grid. If you needed spacing between the grid items you can always add spacing={1} for the container or rowSpacing={1}